src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+    (runtime (linear n u2 u4)))
+
+(define-read-only (cost_analysis_check_tuple_get (n uint))
+    (runtime (logn n u1 u2)))
+
+(define-read-only (cost_analysis_check_tuple_merge (n uint))
+    (runtime (linear n u1000 u1000)))
+
+(define-read-only (cost_analysis_check_tuple_cons (n uint))
+    (runtime (nlogn n u3 u5)))
+
+(define-read-only (cost_analysis_tuple_items_check (n uint))
+    (runtime (linear n u1 u59)))
+
+(define-read-only (cost_analysis_check_let (n uint))
+    (runtime (linear n u1 u12)))
+
+(define-read-only (cost_analysis_lookup_function (n uint))
+    (runtime u20))
+
+(define-read-only (cost_analysis_lookup_function_types (n uint))
+    (runtime (linear n u1 u28)))
+
+(define-read-only (cost_analysis_lookup_variable_const (n uint))
+    (runtime u15))
+
+(define-read-only (cost_analysis_lookup_variable_depth (n uint))
+    (runtime (nlogn n u1 u34)))
+
+(define-read-only (cost_ast_parse (n uint))
+    (runtime (linear n u172 u287441)))
+
+(define-read-only (cost_ast_cycle_detection (n uint))
+    (runtime (linear n u141 u72)))
+
+(define-read-only (cost_analysis_storage (n uint))
+    {
+        runtime: (linear n u2 u100),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+(define-read-only (cost_analysis_use_trait_entry (n uint))
+    {
+        runtime: (linear n u9 u723),
+        write_length: (linear n u1 u1),
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_analysis_get_function_entry (n uint))
+    {
+        runtime: (linear n u81 u1303),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_analysis_fetch_contract_entry (n uint))
+    {
+        runtime: (linear n u1000 u1000),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+(define-read-only (cost_lookup_variable_depth (n uint))
+    (runtime (linear n u2 u14)))
+
+(define-read-only (cost_lookup_variable_size (n uint))
+    (runtime (linear n u2 u1)))
+
+(define-read-only (cost_lookup_function (n uint))
+    (runtime u16))
+
+(define-read-only (cost_bind_name (n uint))
+    (runtime u256))
+
+(define-read-only (cost_inner_type_check_cost (n uint))
+    (runtime (linear n u2 u9)))
+
+(define-read-only (cost_user_function_application (n uint))
+    (runtime (linear n u26 u140)))
+
+(define-read-only (cost_let (n uint))
+    (runtime (linear n u146 u862)))
+
+(define-read-only (cost_if (n uint))
+    (runtime u200))
+
+(define-read-only (cost_asserts (n uint))
+    (runtime u158))
+
+(define-read-only (cost_map (n uint))
+    (runtime (linear n u1210 u3314)))
+
+(define-read-only (cost_filter (n uint))
+    (runtime u460))
+
+(define-read-only (cost_len (n uint))
+    (runtime u486))
```

Comment:
How come this one is cheaper than `filter`?

Reply:
`filter` and `len` are ultimately doing pretty dissimilar things, I'm not sure that I can say definitively one way or another which should be cheaper. The `filter` cost is the cost of the overhead of the `filter` function: checking that its inputs are the appropriate type, dissassembling the list struct to apply the filter function and reassemble an output list. It basically measures everything but the individual invocations of filter. `len` does a lot of the same work, but instead of dissassembling the list struct, it is doing function invocation on the object to get a length. I'm not sure exactly what the difference in runtime would come from here, but I'm not surprised that there is a difference.

Reply:
Thanks for the explanation. 
I thought that cost of `filter` is the cost of all filter operation including individual function invocation on every single element not just the overhead of the `filter`.